### ISSUE TITLE
Brevity is the soul of wit.

### ISSUE THREAD

**Issue by tommyettinger** [SquidPony/SquidLib#61]
I'm really struggling with the direction SquidLib started heading towards around the time of the 2.0 release. I'd be OK with writing slightly more code if the functionality was the same, or only added to... But some things that are actually important to me got removed, and like Steve Ballmer spitting in the face of an amputee, in SquidLib 2.2 I have to go through a layer of enterprise nastiness to do less with more work. Here's what I mean.

In SquidLib 1.95.1, I can ensure that my font will hint correctly (it only hints at sizes that are multiples of 16) and that a glyph will fit exactly in a 12x24 pixel box.
```java
Font font = Font.createFont(Font.TRUETYPE_FONT, new File("my-font.ttf")).deriveFont(32.0);
SwingPane sp = new SwingPane(12, 24, 60, 60, font);
```

In SquidLib 2.2, I might be able to make this work with fitting and padding, but I still don't know if the font will be resized by TextCellFactory.
```java
Font font = Font.createFont(Font.TRUETYPE_FONT, new File("my-font.ttf")).deriveFont(32.0);
TextCellFactoryBuilder uss = new TextCellFactoryBuilder().font(font).width(12).height(24);
TextCellFactory enterprise = new TextCellFactory(uss);
SquidPanel sp = new SquidPanel(60, 60, enterprise, null);
```

Is TextCellFactoryBuilder even needed?  It provides almost a carbon copy of the structure of TextCellFactory, and all of the methods it has would be perfectly suited to just being setters in TextCellFactory.  It's an excellent example of why I am so deeply opposed to this sort of enterprise code; it violates the DRY principle, it's more verbose than is warranted by the problem, and it hides capabilities that should be accessible (and were in earlier versions).

I'm porting some C# code that uses [BearLibTerminal](http://foo.wyrd.name/en:bearlibterminal) to Clojure, and I was hoping to use SquidLib's most recent features, but now I'm using 1.95.1. BearLibTerminal has 27 functions and some constants declared in one namepace, and that's it. It doesn't do anything other than display, but it handles images, multiple fonts, adjustments to position within a cell (bump or slide effects use this), and supports both bitmap and truetype fonts. I strongly prefer its style of API to any API that involves multiple closely-linked classes, where I need to go between 3 different doc pages to see what I need to do to use one of them. Does this make sense?

**Comment by SquidPony** [comment]
TL:DR - I mostly agree with this, but it ended up with a Builder for what seemed like good reasons at the time. It's entirely possible however that a different approach would be better. One of those reasons is that the sizing in the factory needs all the details before it sizes and making setters implies that those things can be changed after the factory is created. This would probably be fine if it wasn't so computationally expensive to make a factory though.

Some of these concerns were meant to be addressed by a bootstrap class that would basically be the default implementation of everything with a standard foreground and background, basically getting back to the interface of 1.95 without losing the more powerful options now available. This became fairly awkward while working on it though and the use cases for it seemed thin so I yanked it back out.

Looking at BearLibTerminal has been on my todo list for a bit, I'll bump it up to a higher priority.

But first, I'm going to take a fresh look at the state of the panel factory builder stuff and see what's what because I have a feeling that there is indeed some room for improvement now that I've got some distance from it.

**Comment by SquidPony** [comment]
Oh, and to answer the question regarding font size, the way you have it there should give exactly the same output as previously. If it doesn't then that's a bug in either 1.95.1 or 2.2 (or both).

There could be an option to force font size, even if it might be too large theoretically, but that's adding to the  Builderness of it that's a concern.

**Comment by SquidPony** [comment]
Okay, Phase 1 cleanup already, removed TextCellFactoryBuilder, pushing its functionality into TextCellFactory and streamlining a bit with SquidPanel's constructors.

I'll close this once I make some issues for the other things brought up.

**Comment by tommyettinger** [comment]
I think uh... I think the issue raised here has been entirely resolved. In part because now I've written most of SquidLib, so any remaining issues are probably my fault. In fact I've probably changed enough of TextCellFactory for just stuff like distance field fonts that I'm now well-familiar with the text layout code. This issue is like a portal back through time... Time to close it. Sorry for the nasty tone earlier, Dr. Roguelike!
